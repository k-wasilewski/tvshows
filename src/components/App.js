import React from 'react';
import Router from './Router';
import {useStyles} from '../styles/styles';
import logo from '../img/logo.png';

export default function App() {
    const classes = useStyles();

    return (
        <div className={classes.App}>
            <img className={classes.logo} src={logo} />
            <Router />
            <p className={classes.copywright}> &copy; Kuba Wasilewski, 2020 </p>
        </div>
    );
};
