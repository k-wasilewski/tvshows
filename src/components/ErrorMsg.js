import React from 'react';
import {useStyles} from "./useStyles";

export function ErrorMsg(props) {
    const classes = useStyles();

    return (
        <span className={classes.errorMsg}>{props.msg}</span>
    )
}