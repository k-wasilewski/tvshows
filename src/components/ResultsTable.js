import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import { useHistory } from "react-router-dom";
import {setDetailedResult} from "../redux/actions";
import {connect} from "react-redux";
import {useStyles, BACKGROUND_COLOR} from "./useStyles";

const mappedGenres = (genres) => {
    if (genres.length===1) return genres;
    else {
        return genres.map(genre => {
            if (genres.indexOf(genre)!==genres.length-1)
                return genre+=', ';
            else return genre;
        });
    }
}

const StyledTableHeadRow = withStyles((theme) => ({
    root: {
        '& *': {
            fontFamily: 'Lato Heavy',
            fontSize: '1.1rem'
        }
    }
}))(TableRow);

const StyledTableRow = withStyles((theme) => ({
    root: {
        '&:hover': {
            backgroundColor: 'lightgray',
            cursor: 'pointer'
        }
    }
}))(TableRow);

const StyledTable = withStyles((theme) => ({
    root: {
        backgroundColor: BACKGROUND_COLOR
    }
}))(Table);

export function ResultsTable(props) {
    const classes = useStyles();
    const [sortedResults, setSortedResults] = React.useState([]);

    const history = useHistory();

    function rowClicked(event, result) {
        props.setDetailedResult(result);
        history.push("/details");
    }

    React.useEffect( () => {
        setSortedResults(props.results.sort((a, b) => b.score - a.score));
    }, [props.results])

    return (
        <TableContainer className={classes.tableWrapper} component={Paper} id='tableWrapper'>
            <StyledTable aria-label="results table">
                <TableHead>
                    <StyledTableHeadRow>
                        <TableCell>Ocena</TableCell>
                        <TableCell align="right">Tytu≈Ç</TableCell>
                        <TableCell align="right">Gatunki</TableCell>
                        <TableCell align="right">Data premiery</TableCell>
                    </StyledTableHeadRow>
                </TableHead>
                <TableBody>
                    {sortedResults.map((result) => (
                        <StyledTableRow key={result.show.name} onClick={event => rowClicked(event, result)}>
                            <TableCell component="th" scope="row">{result.score}</TableCell>
                            <TableCell align="right">{result.show.name}</TableCell>
                            <TableCell align="right">{mappedGenres(result.show.genres)}</TableCell>
                            <TableCell align="right">{result.show.premiered}</TableCell>
                        </StyledTableRow>
                    ))}
                </TableBody>
            </StyledTable>
        </TableContainer>
    );
}

const mapStateToProps = state => ({
    detailedResult: state.setDetailedResultReducer.detailedResult
});

const mapDispatchToProps = { setDetailedResult };

export default connect(mapStateToProps, mapDispatchToProps)(ResultsTable);