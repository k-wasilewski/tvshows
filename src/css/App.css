.App {
  text-align: center;
  font-family: Lato;
}

#errorMsg {
  color: red;
}

#copyright {
  bottom: 10px;
  position: fixed;
  right: 10%;
}
