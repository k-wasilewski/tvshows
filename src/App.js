import React from 'react';
import Router from "./components/Router";
import {useStyles} from "./components/useStyles";

export default function App() {
    const classes = useStyles();

    return (
        <div className={classes.App}>
            <Router />
        </div>
    );
};
