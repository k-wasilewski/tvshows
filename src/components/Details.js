import React, { Component } from 'react';
import '../css/App.css';
import {setDetailedResult} from "../redux/actions";
import {connect} from "react-redux";
import {Link} from "react-router-dom";

class Details extends Component {

    createMarkup = (text) => {
        return {
            __html: text
        };
    };

    render() {
        const result = this.props.detailedResult;

        if (result.length===0) {
            return (<div className='errorMsg'>Należy wybrać element z listy</div>)
        } else {
            return (
                <div className="App">
                    <h2>{result.show.name}</h2>
                    <span dangerouslySetInnerHTML={this.createMarkup(result.show.summary)} />
                    <div><img alt={`img-${result.show.name}`} src={result.show.image.medium} /></div>
                    <Link to="/"><button>Powrót</button></Link>
                </div>
            );
        }
    };
};

function mapStateToProps(state) {
    return {
        detailedResult: state.setDetailedResultReducer.detailedResult
    };
};

const mapDispatchToProps = {
    setDetailedResult
};

export default connect(mapStateToProps, mapDispatchToProps)(Details);
