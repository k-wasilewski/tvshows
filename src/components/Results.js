import React, { Component } from 'react';
import '../css/App.css';
import {setResults} from "../redux/actions";
import {connect} from "react-redux";
import ResultsTable from "./ResultsTable";
import ResultsFilter from "./ResultsFilter";
import mapDayNoToName from "../functions/mapDayNoToName";

class Results extends Component {
    constructor(props) {
        super(props);

        this.state = {
            filteredResults: []
        };
    }

    filterByDay = (day) => {
        const dayName = mapDayNoToName(day);

        const filteredResults = this.props.results.filter(result =>
            result.show.schedule.days.includes(dayName)
        );

        this.setState({filteredResults: filteredResults});
    }

    render() {
        return (
            <div className="App">
                <ResultsFilter setDay={this.filterByDay} />
                <ResultsTable results={(this.state.filteredResults.length===0) ?
                    this.props.results
                    :
                    this.state.filteredResults} />
            </div>
        );
    };
};

function mapStateToProps(state) {
    return {
        results: state.setResultsReducer.results,
    };
};

const mapDispatchToProps = {
    setResults
};

export default connect(mapStateToProps, mapDispatchToProps)(Results);
