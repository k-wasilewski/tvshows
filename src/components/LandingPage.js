import React, { Component } from 'react';
import '../css/App.css';
import Search from './Search';
import Results from './Results';

class LandingPage extends Component {
    render() {
        return (
            <div className="App">
                <Search />
                <Results />
            </div>
        );
    };
};

export default LandingPage;
