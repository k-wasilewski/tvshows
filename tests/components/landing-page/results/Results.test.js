import React from 'react';
import renderer from 'react-test-renderer';
import store from "../../../../src/redux/store";
import {BrowserRouter} from "react-router-dom";
import ConnectedResults, {Results} from "../../../../src/components/landing-page/results/Results";
import {Provider} from "react-redux";
import {configure} from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import {mount, shallow} from "enzyme";
import {App} from "../../../../src/components/App";

describe("Results rendering specification", () => {
    it('Results is rendered', () => {
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <ConnectedResults/>
                </BrowserRouter>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("Results functional specification", () => {

    it('clears the state.filteredResults when componentDidUpdate with unchanged ' +
        'state.filteredResults value', () => {
        configure({adapter: new Adapter()});

        const testFilteredResultsValue = {
            show: {
                name: 'sample name',
                genres: ['sample genre']
            },
            score: 2
        };
        const testChangedFilteredResultsValue = {
            show: {
                name: 'changed',
                genres: ['sample genre']
            },
            score: 3
        };

        const component = mount(
            <Provider store={store}>
                <Results results={[testFilteredResultsValue]}/>
            </Provider>
        );

        component.find(Results).prop('results').sort((a, b) => a-b)
        component.find(Results).setState({filteredResults: testFilteredResultsValue});
        console.log(Array.isArray(component.find(Results).prop('results')))
        component.update();
        expect(component.find(Results).state('filteredResults')).toBe(testFilteredResultsValue);
        component.find(Results).setState({filteredResults: testChangedFilteredResultsValue});
        component.setProps({children: <Results results={[testChangedFilteredResultsValue]}/>});
        component.update();
        expect(component.find(Results).state('filteredResults')).toStrictEqual([]);
        component.unmount();
    });

    it('filterByDay() filters results from props to state.filteredResults value ' +
        'by passed day number', (done) => {
        configure({adapter: new Adapter()});

        const resultMonday = {
            show: {
                name: 'sample name',
                genres: ['sample genre'],
                schedule: {
                    days: 'Monday'
                }
            },
            score: 2
        };

        const resultTuesday = {
            show: {
                name: 'sample name',
                genres: ['sample genre'],
                schedule: {
                    days: 'Tuesday'
                }
            },
            score: 2
        };

        const testInitialResults = [resultMonday, resultTuesday];

        const component = shallow(
            <Results results={testInitialResults}/>
        );

        component.instance().filterByDay(1);
        component.update();

        setTimeout(function () {
            expect(JSON.stringify(component.state('filteredResults')))
                .toBe(JSON.stringify([resultMonday]));
            component.unmount();
            done();
        }, 500);
    });
});