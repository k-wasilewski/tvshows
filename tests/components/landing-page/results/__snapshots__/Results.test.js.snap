// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results rendering specification Results is rendered 1`] = `null`;
